packages/corejs/src/monitoring/index.tsx

```diff
@@ -0,0 +1,128 @@
+const setTag = (key: string, value: Primitive): void => SentryR.setTag(key, value);
+const setUser = (id: string) => SentryR.setUser({ id });
+const MonitoringContext = createContext<MonitoringContextType | null>(null);
+
+function MonitoringProvider({ children, config }) {
+  useEffect(() => {
+    Sentry.init({
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      dsn: (config.dns || Constants.manifest.extra.SENTRY_DSN) as string,
+      environment: Constants.manifest.extra.ENV as string,
+      debug: Constants.manifest.extra.ENV !== 'production',
+      enableAutoSessionTracking: true,
+      enableInExpoDevelopment: true,
+      sessionTrackingIntervalMillis: 10000,
+      tracesSampleRate: Constants.manifest.extra.ENV !== 'production' ? 1.0 : 0.2,
+      integrations: [
+        ...(Platform.select({
+          web: [new CaptureConsole({ levels: ['log'] })],
+          native: [],
+        })),
+        new Tracing({
+          tracingOrigins,
+        }),
+      ],
+    } as Sentry.SentryExpoNativeOptions);
+  }, []);
+
+  /**
+   * @param error {Error} - Caught error to send to Sentry.io
+   * @param type {string} - tag of the error's type
+   * @param extras {Record} - Useful information that can be sent (request's body for example)
+   * @param additionalTags - (Optional) Additional tags to add to the error
+   */
+  const errorHandler = (error: Error | string): string => {
+    if (!Sentry || (!Sentry?.Browser && !Sentry?.Native)) { return null; }
+
+    return Platform.select({
+      web: Sentry.Browser,
+      native: Sentry.Browser,
+    }).captureException(error);
+  };
+
+  /**
+   * @param name {string} - Name of transaction
+   * @param operation {string} - Operation of transaction to be performed
+   * @param data {{[key: string]: number | string}} - Data to be added on transaction
+   */
+  const measurePerformance = (name: string, operation: string, data: { [key: string]: number | string } | null) => {
+    const transaction = Platform.select({
+      web: Sentry.Browser,
+      native: Sentry.Browser,
+    }).startTransaction({ name, data });
+    const transactionOperation = transaction.startChild({ op: operation });
+
+    return () => {
+      transactionOperation.finish();
+      transaction.finish();
+    };
+  };
+
+  const monitoringContextValue = useMemo(
+    () => ({ errorHandler, measurePerformance }),
+    [errorHandler, measurePerformance],
+  );
+
+  return (
+    <MonitoringContext.Provider value={monitoringContextValue}>
+      {children}
+    </MonitoringContext.Provider>
+  );
+}
+
+const useMonitoring = () => useContext(MonitoringContext);
+
+export {
+  MonitoringConstants,
+  MonitoringContext,
+  setTag,
+  setUser,
+  Tracing,
+  MonitoringProvider,
+  useMonitoring,
+};
+
+MonitoringProvider.propTypes = {
+  config: PropTypes.shape({
+    dns: PropTypes.string,
+  }),
+};
+
+MonitoringProvider.defaultProps = {
+  config: {
+    dns: '',
+  },
+};
```

Comment:
Since we are in TypeScript I don't think we need PropTypes, we can actually use TypeScript type defs which is better